File: CS/CustomLogonParametersExample.Module/BusinessObjects/Company.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace CustomLogonParametersExample.Module.BusinessObjects {
    [DefaultClassOptions]
    public class Company : BaseObject {
        public Company(Session session) : base(session) { }
        private string name;
        public string Name {
            get { return name; }
            set { SetPropertyValue("Name", ref name, value); }
        }
        [Association("Company-Employees", typeof(Employee))]
        public XPCollection Employees {
            get { return GetCollection("Employees"); }
        }
    }

}

File: VB/CustomLogonParametersExample.Module/BusinessObjects/Company.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace CustomLogonParametersExample.Module.BusinessObjects
	<DefaultClassOptions, ImageName("BO_Organization")> _
	Public Class Company
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private name_Renamed As String
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
		<Association("Company-Employees", GetType(Employee))> _
		Public ReadOnly Property Employees() As XPCollection
			Get
				Return GetCollection("Employees")
			End Get
		End Property
	End Class

End Namespace
